crates/vm2/src/stack.rs

```diff
@@ -19,6 +19,7 @@ const NUMBER_OF_DIRTY_AREAS: usize = 64;
 const DIRTY_AREA_SIZE: usize = (1 << 16) / NUMBER_OF_DIRTY_AREAS;
 
 impl Stack {
+    #[allow(clippy::cast_ptr_alignment)] // aligned per `Stack` layout
```

Comment:
Maybe switching Stack to Self satisfies the lint?

Reply:
No, it doesn't. `alloc_zeroed` returns `*mut u8`, which has lower alignment requirements compared to `Layout`. clippy isn't smart enough to figure out that the actual alignment depends on the arg supplied to `alloc_zeroed` (I mean, it's very difficult / impossible to figure out in the general case).